lib/Common/Core/SysInfo.cpp

```diff
@@ -8,6 +8,7 @@
 #endif
 #include <wincrypt.h>
 #include <VersionHelpers.h>
+#include <type_traits>
```

Comment:
Fails for cross platform.

I guess the `AutoSystemInfoInitializer` is process wide. Instead of dealing with counter, can we use the same static initialization approach (which is also thread safe) ? 

- Is there a reason why don't we host this under PAL ? 

- Why we do limit this only for static library?